torch/utils/data/_utils/worker.py

```diff
@@ -9,12 +9,31 @@
 import queue
 import random
 from dataclasses import dataclass
-from typing import Optional, TYPE_CHECKING, Union
+from typing import Any, Optional, TYPE_CHECKING, TypeVar, Union
 
 import torch
 from torch._utils import ExceptionWrapper
 
 from . import HAS_NUMPY, IS_WINDOWS, MP_STATUS_CHECK_INTERVAL, signal_handling
+from .stateful import Stateful
+
+
+T = TypeVar("T")
+
+
+# Stateful functionality
+def _try_get_state_dict(obj: Any) -> Optional[dict[str, Any]]:
+    """Try to get state_dict from an object if it implements Stateful protocol."""
+    if isinstance(obj, Stateful):
+        return obj.state_dict()
+    return None
+
+
+def _try_load_state_dict(obj: T, state_dict: dict[str, Any]) -> None:
```

Comment:
nit: The `T` doesn't serve any purpose here -- it's not constrained, and it doesn't constrain any other types in the method's signature.

```suggestion
def _try_load_state_dict(obj: Any, state_dict: dict[str, Any]) -> None:
```